graph-partitioning/README.rst

```diff
@@ -0,0 +1,82 @@
+Graph Partitioning Demo
+=======================
+In this demo, we explore the graph partitioning problem.  This is an interesting problem from graph theory with a wide variety of applications (see e.g. [3]_).
+
+For example, suppose that we have a group of people that we need to split into two equal-sized teams.  For each pair of people there is a relationship:  friends or enemies.  We want to have as few friends on opposite teams as possible.  How can we solve this problem?
+
+One way to solve this problem is with the graph partitioning problem.  If we think of our group of people as a graph (a node/vertex for each person), and draw an edge between people that are friends, we have a model of our network.  If we look for a minimum cut in our graph, then we are looking for a way to split the nodes into two groups so that there are as few edges as possible between the groups.  In our group of peoople, this means that we have two teams with as few friends as possible on different teams.
+
+We will run the graph partitioning problem on an Erdos-Renyi random graph with probability 0.2 on 40 nodes to find the best way of splitting the graph into two subsets to minimize the number of cut edges.
+
+Usage
+-----
+To run the demo, type:
+::
+  python graph_partitioning.py
+
+After running, output will be printed to the command line that states whether or not the best (lowest energy) solution found is a valid partition of two equal-sized sets.  If the solution is valid (feasible), then the number of cut edges for that solution is displayed.
+
+Code Overview
+-------------
+The code implements a QUBO formulation of this problem.
+
+The answer that we are looking for is a partition of the nodes in the graph, so we will assign a binary variable for each node, i.e. variable x_i denotes whether node i is in one subset (call it Subset 0) or the other (Subset 1).
+
+The objective function that we are looking to optimize is minimizing the number of cut edges.  To count how many cut edges we have given a partition of the nodes (assignment of our binary variables), we consider a single edge in a graph in the table below.  We only want to count an edge if the endpoints are in different subsets, and so we assign a 1 for the edge column in this case and a 0 otherwise.
+
+=== === ==========
+x_i x_j edge (i,j)
+--- --- ----------
+0   0   0
+0   1   1 
+1   0   1
+1   1   0
+=== === ==========
+
+From this table, we see that we can use the expression x_i+x_j-2x_ix_j to calculate the edge column in our table.  Now for our entire graph, our objective function can be written as shown below, where the sum is over all edges in the graph.
+
+.. image:: readme_imgs/QUBO.png
+
```

Comment:
RST files support math formatting so why use a picture? In fact we try to avoid that where we can so users with visual impairments can use text-to-speech software.

Reply:
Also, `E` in the sum is not defined

Reply:
When I use the RST math formatting it writes everything in-line, which gets messy when we start doing summations.  @m3ller which would you prefer?  See screenshot with both side by side.
<img width="407" alt="Screen Shot 2019-08-19 at 2 46 03 AM" src="https://user-images.githubusercontent.com/38987205/63228278-97f04900-c22b-11e9-8958-5a2badba8a09.png">

Reply:
@vgoliber, here's how you can format it on a standalone line:

.. math::
  \sum_

Reply:
That's exactly what I did to produce the top line - take a look at the updated RST file.  In raw form you should see that I typed it in that way but it doesn't display the math nicely.

Reply:
Oh, I see what's happening, the math is not being rendered by github (inc inline), apparently they don't run mathjax.

Reply:
Unfortunately, it doesn't seem like GitHub RST supports `math`. It seems like they would have issues with securely deploying such a request.
https://github.com/github/markup/issues/83
https://github.com/github/markup/issues/274

The picture workaround seems to give the best user experience.

Reply:
Yes, that's why I did the picture in the first place.

I'll remove the inline expression.